### PR TITLE
Added Support for Headers with Tile Fetches

### PR DESCRIPTION
**Pull request by theduckylittle**
1. Checks for an Authorization header in fetchOptions.
2. When present, uses an async loader to get the data
   using fetch, convert it to a BLOB and then set the Tile's
   image to that URL.

### CODE REVIEW

File: src/components/map.js [hunk 1]
```diff
@@ -125,15 +125,39 @@ function getVersion(obj, key) {
   return obj.metadata[key];
 }
 
+/**
+ * Load a tile with full fetchptions.
```
Comment (bartvde) [inline src/components/map.js]: fetchOptions

File: src/components/map.js [hunk 2]
```diff
@@ -125,15 +125,39 @@ function getVersion(obj, key) {
...
+ * @param {Object} fetchOptions Options to use for fetch calls.
+ *
+ * @returns {Function} Tile loading function which uses fetch.
+ */
+function authTileLoader(fetchOptions) {
+  return function(tile, src) {
+    fetch(src, fetchOptions)
+      .then(r => r.blob())
+      .then((imgData) => {
+        tile.getImage().src = URL.createObjectURL(imgData);
```
Comment (bartvde) [inline src/components/map.js]: we probably need a polyfill for URL?

File: src/components/map.js [hunk 3]
```diff
@@ -125,15 +125,39 @@ function getVersion(obj, key) {
...
+      })
+      .catch(() => {
+        console.error('Error fetchimg image at:', this.props.src);
```
Comment (bartvde) [inline src/components/map.js]: fetching

Comment (bartvde) [reply src/components/map.js]: this.props.src or simply src?

File: src/components/map.js (hunk 2 above)
Comment (theduckylittle) [reply src/components/map.js]: Do we? https://caniuse.com/#feat=bloburls

Comment (bartvde) [reply src/components/map.js]: ah right, seems we don't need it

**Pull request merged**